curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -2031,20 +2064,20 @@ CURVEFS_ERROR DataCache::Flush(uint64_t inodeId, bool force, bool toS3) {
         PutObjectAsyncCallBack cb =
             [&](const std::shared_ptr<PutObjectAsyncContext> &context) {
                 if (context->retCode == 0) {
+                    if (pendingReq.fetch_sub(
```

Comment:
Don't move the if sentence to the front，it will cause core dumped because s3Metric_ will be destructed before being accessed, see pr : https://github.com/opencurve/curve/pull/1184

Reply:
By the way, adding a comment here in case someone else changes it

Reply:
fix